corehq/form_processor/management/commands/fix_edit_forms_attachments.py

```diff
@@ -0,0 +1,121 @@
+        parser.add_argument('--inspect', action='store_true', help='just write findings to file')
+
+    @staticmethod
+    def _find_sql_forms_with_missing_attachments():
+        """
+        find all forms that are a result of a form edition
+        find the deprecated form and then compare attachments between them
+        :return:
+        """
+        add_forms_with_attachments = defaultdict(list)
+        form_accessor = FormAccessorSQL()
+        for db_alias in get_db_aliases_for_partitioned_query():
+            print('scanning %s' % db_alias)
+            edit_form_ids = (XFormOperationSQL.objects.using(db_alias)
+                             .filter(operation='edit')
+                             .values_list('form_id', flat=True))
+            for edit_form_id in with_progress_bar(edit_form_ids):
+                edited_form = form_accessor.get_form(edit_form_id)
+                deprecated_form_id = edited_form.deprecated_form_id
+                original_attachments = list(FormAccessorSQL.get_attachments_for_forms([deprecated_form_id]))
+                new_attachments = list(FormAccessorSQL.get_attachments_for_forms([edit_form_id]))
+                new_attachment_names = [a.name for a in new_attachments]
+                for attachment in original_attachments:
+                    if attachment.name not in new_attachment_names:
+                        assert attachment.form_id == deprecated_form_id
+                        add_forms_with_attachments[edit_form_id].append(
+                            (attachment.attachment_id, deprecated_form_id)
+                        )
+        return add_forms_with_attachments
+
+    @staticmethod
+    def _find_couch_forms_with_missing_attachments():
+        """
+        find deprecated form ids,
+        then look for the new form and compare attachments between them
+        :return:
+        """
+        add_forms_with_attachments = defaultdict(list)
+        deprecated_form_ids = [
+            f['id'] for f in XFormDeprecated.view(
+                'all_docs/by_doc_type', startkey=['XFormDeprecated'], endkey=['XFormDeprecated', {}],
+                include_docs=False, reduce=False).all()]
+        for deprecated_form_id in deprecated_form_ids:
+            deprecated_form = XFormDeprecated.get(deprecated_form_id)
+            form_accessor = FormAccessors(deprecated_form.domain)
+            if deprecated_form.orig_id:
+                try:
+                    new_form = form_accessor.get_form(deprecated_form.orig_id)
+                except XFormNotFound:
+                    # fall back to other DB couch/sql to look for the form
+                    if form_accessor.db_accessor == FormAccessorSQL:
+                        new_form = FormAccessorCouch.get_form(deprecated_form.orig_id)
+                    else:
+                        new_form = FormAccessorSQL.get_form(deprecated_form_id)
+                except ValueError as e:
```

Comment:
I have then printed and it was always that error so i am sure its catching only what i expect it to

Reply:
That may be true in tests you've done so far. I would not assume that for all forms this may eventually run on.

Reply:
I also dont want the loop to halt, so whatever the error is, just print and continue. I will just add the form id so that it can be looked later if needed.